vnext/Microsoft.ReactNative.Cxx/DesktopWindowBridge.h

```diff
@@ -0,0 +1,105 @@
+}
+} // namespace details
+
+inline void ForwardWindowMessage(
+    const ReactNotificationService &svc,
+    HWND hwnd,
+    uint32_t msg,
+    WPARAM wParam,
+    LPARAM lParam) noexcept {
+  DesktopWindowMessage args{msg, wParam, lParam};
+  svc.SendNotification(details::GetNotificationIdForWindowMessage(msg), winrt::box_value((uint64_t)hwnd), args);
+}
+
+inline ReactNotificationSubscription SubscribeToWindowMessage(
+    const ReactNotificationService &svc,
+    uint32_t msg,
+    std::function<void(HWND, const DesktopWindowMessage &)> &&callback) noexcept {
+  return svc.Subscribe(
+      details::GetNotificationIdForWindowMessage(msg),
+      [cb = std::move(callback)](
+          const winrt::Windows::Foundation::IInspectable &sender,
+          const ReactNotificationArgs<DesktopWindowMessage> &args) {
+        if (auto dwm = args.Data()) {
+          cb(reinterpret_cast<HWND>(winrt::unbox_value<uint64_t>(sender)), dwm);
+        }
+      });
+}
+
+inline std::unordered_map<uint32_t, ReactNotificationSubscription> SubscribeToWindowMessage(
+    const ReactNotificationService &svc,
+    std::initializer_list<uint32_t> msgs,
+    std::function<void(HWND, const DesktopWindowMessage &)> &&callback) noexcept {
+  std::unordered_map<uint32_t, ReactNotificationSubscription> result;
+  for (auto msg : msgs) {
+    result[msg] = SubscribeToWindowMessage(svc, msg, std::move(callback));
+  }
+  return result;
+}
+
+#else
+namespace winrt::Microsoft::ReactNative {
+
+/// Stubs for UWP
+template <typename... T>
+void ForwardWindowMessage(const ReactNotificationService &svc, T &&...) noexcept {}
+
+template <typename... T>
+ReactNotificationSubscription SubscribeToWindowMessage(const ReactNotificationService &svc, T &&...) noexcept {
+  return nullptr;
+}
+
+template <typename... T>
+std::unordered_map<uint32_t, ReactNotificationSubscription>
+SubscribeToWindowMessage(const ReactNotificationService &svc, std::initializer_list<uint32_t> msgs, T &&...) noexcept {
+  return std::unordered_map<uint32_t, ReactNotificationSubscription>{};
+}
+#endif
+
+namespace details {
+struct IndirectLibraryDeleter {
+  void operator()(void *l) {
+    WINRT_IMPL_FreeLibrary(l);
+  }
+};
+
+using IndirectLibrary = std::unique_ptr<void, IndirectLibraryDeleter>;
+__declspec(selectany) std::unordered_map<std::wstring, IndirectLibrary> indirectLibraries{};
+} // namespace details
+
+template <typename TFn, typename... TArgs>
+auto CallIndirect(const wchar_t *dllName, const char *fnName, TArgs &&... args) noexcept {
+  if (details::indirectLibraries.count(dllName) == 0) {
+    details::indirectLibraries.emplace(dllName, WINRT_IMPL_LoadLibraryW(dllName));
+  }
+  auto &library = details::indirectLibraries[dllName];
+  auto pfn = reinterpret_cast<TFn>(WINRT_IMPL_GetProcAddress(library.get(), fnName));
+  return pfn(args...);
+}
+
+#define CALL_INDIRECT(dllName, fn, ...) \
+  winrt::Microsoft::ReactNative::CallIndirect<decltype(&fn)>(dllName, #fn, __VA_ARGS__)
```

Comment:
>__VA_ARGS__ [](start = 75, length = 11)

This will probably not work for functions without arguments

Reply:
seems to work: 
![image](https://user-images.githubusercontent.com/22989529/108182002-69238f00-70bd-11eb-9d8f-c80f26947818.png)